ziti/cmd/ziti/cmd/fabric/add_identity.go

```diff
@@ -0,0 +1,82 @@
+/*
+	Copyright NetFoundry, Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package fabric
+
+import (
+	"github.com/openziti/ziti/ziti/cmd/ziti/cmd/common"
+	"github.com/openziti/ziti/ziti/cmd/ziti/util"
+	"github.com/spf13/cobra"
+)
+
+type addIdentityOptions struct {
+	common.CommonOptions
+	caCert     string
+	clientCert string
+	clientKey  string
+	readOnly   bool
+}
+
+func newAddIdentityCmd(p common.OptionsProvider) *cobra.Command {
+	options := &addIdentityOptions{
+		CommonOptions: p(),
+	}
+
+	cmd := &cobra.Command{
+		Use:   "add-identity <server-url>",
+		Short: "adds a fabric identity for connecting to the Ziti Controller",
```

Comment:
is a fabric identity different than an 'edge' identity different than a 'controller' identity?

Reply:
A fabric identity is signed client cert recognized by the controller. An edge identity is an api session token. The controller itself has an identity, which consists of a cert chain.